third_party/move/move-vm/runtime/src/runtime.rs

```diff
@@ -304,13 +305,7 @@ impl VMRuntime {
 
         let (layout, has_identifier_mappings) = self
             .loader
-            .type_to_type_layout_with_identifier_mappings(ty, module_store)
-            .map_err(|_err| {
-                // TODO: Should we use `err` instead of mapping?
-                PartialVMError::new(StatusCode::VERIFICATION_ERROR).with_message(
-                    "entry point functions cannot have non-serializable return types".to_string(),
-                )
-            })?;
+            .type_to_type_layout_with_identifier_mappings(ty, module_store)?;
```

Comment:
This either needs to be
1. Feature-gated
2. Reverted (better to do it in a separate PR)

Let's do (2) as we discussed offline

Reply:
Fixed